pkg/reconciler/namespace/conditions_test.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package namespace
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	conditionsapi "github.com/kcp-dev/kcp/third_party/conditions/apis/conditions/v1alpha1"
+	"github.com/kcp-dev/kcp/third_party/conditions/util/conditions"
+)
+
+func TestSetScheduledCondition(t *testing.T) {
+	testCases := map[string]struct {
```

Comment:
(another fun non-blocking style nit comment!) I never really see this phrased in terms of a `map[string]struct`, usually just a `[]struct` with some description field. I suppose it's roughly equivalent, except that execution order will be randomized using a map. I don't think that would be a problem in this case, but I think that's probably why the idiomatic way is a `[]struct` (and maybe saving some indentation?).

Reply:
I've gotten into the habit of using maps by default to avoid any dependency on test ordering. Tests relying on order of execution are by definition not hermetic, which is generally an anti-pattern.

I think you'll find there's effectively no difference in indentation from struct to map:


```golang
testCases := []struct{
   name string
   bar     string
} {
    {
        name: "foo",
        bar:     "baz",
    },
}

testCases := map[string]struct{
    foo string
} {
    "foo": {
        bar: "baz",
    },
}
```

Reply:
> I think you'll find there's effectively no difference in indentation from struct to map:

True enough, unless you collapse the indentation (like I tend to do), which is a bit harder to justify with a map:

```go
testCases := []struct{
    name string
    bar  string
} {{
    name: "foo",
    bar:  "baz",
}}
 
testCases := map[string]struct{
    foo string
} {
    "foo": {
        bar: "baz",
    },
}
```

Anyway, I've spent my per-PR budget on style nits. Resolving.

Reply:
This is the first time I'm seeing the indentation being collapsed, now I see what you mean!

Reply:
For bonus compactness points, I usually do this:

```golang
for _, c := range []struct {
  desc string
  ...
  want string
} {{
  desc: "first",
  want: "whatever",
}, {
  desc: "second",
  want: "another thing",
}} {
  t.Run(c.desc, func(t *testing.T) {
    ...
  }
}

```

Reply:
I think I'm convinced to use this compact form for unit tests that have no potential interaction. Maybe for e2e where shared fixture is involved, maps would be worth the loss of compactness.